### PR TITLE
8.0 pxc 2451

### PR DESCRIPTION
**Pull request by kennt-percona**

### CODE REVIEW

**Comment by kennt-percona** [comment]
https://jenkins.percona.com/job/pxc80.build/113/

File: scripts/wsrep_sst_common.sh [hunk 1]
```diff
@@ -731,7 +989,7 @@ EOF
         --defaults-file=/dev/stdin \
         --socket=$upgrade_socket \
         --unbuffered --batch --silent \
-        -e "SET sql_log_bin = OFF; ALTER USER '${WSREP_SST_OPT_USER}'@localhost ACCOUNT LOCK; SHUTDOWN;" \
+        -e "SET sql_log_bin=OFF; ALTER USER '${WSREP_SST_OPT_USER}'@localhost ACCOUNT LOCK; SHUTDOWN;" \
```
Comment (kbauskar) [inline scripts/wsrep_sst_common.sh:1002]: please add comment why we need to lock the user before shutdown when on SHUTDOWN action we tend to remove the user.
(as we discussed during call a defensive check).

File: scripts/wsrep_sst_common.sh [hunk 2]
```diff
@@ -430,6 +528,41 @@ function check_for_version()
 #   0 if successful (no errors encountered)
 #   <non-zero> otherwise
 #
+# Overview: There are two code paths:
+#   (1) Receiving from a < 8.0 donor (5.7 donor)
+#   (2) Receiving from a >= 8.0 donor
+#
+#   (1) Receiving from a 5.7 donor
```
Comment (kbauskar) [inline scripts/wsrep_sst_common.sh:543]: Can you expand this comment to make it more explicit that 8.0 binaries will not allow CREATE USER TO work on DONOR copied 5.7 data-directory.

**Review by kbauskar — CHANGES_REQUESTED** [review]
Can you also change the user from mysql.pxc.sst.root to mysql.pxc.internal.session.user

**Comment by kbauskar** [comment]
@kennt-percona 
Can you also add force-upgrade to default mtr configuration include/default_mysqld.cnf

File: scripts/wsrep_sst_common.sh (hunk 2 above)
Comment (kennt-percona) [reply scripts/wsrep_sst_common.sh:543]: There's a comment below (with the detailed flow) that explains this.

File: scripts/wsrep_sst_common.sh (hunk 1 above)
Comment (kennt-percona) [reply scripts/wsrep_sst_common.sh:1002]: comment added

**Review by kbauskar — APPROVED** [review]

**Pull request merged**